hal_silabs: Add support for librail for 1x devices

yonsch opened the pull request:
Update hal_silabs to support the use of librail for efr32xg1x devices.

yonsch committed:
west: hal_silabs: Add support for librail for 1x devices

Update hal_silabs to support the use of librail for efr32xg1x
devices.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>

yonsch commented:
ref: https://github.com/zephyrproject-rtos/hal_silabs/pull/35

fkokosinski commented:
Hi, thanks for this PR. It looks like:
1. you need to create "binary blobs integration" issue using the template from here: https://github.com/zephyrproject-rtos/zephyr/issues/new/choose
2. no zephyr platform uses these blobs right now (no bluetooth changes etc.)?

yonsch commented:
@fkokosinski Thanks, the issue is here: #61109 
I'm using RAIL without BT, so I haven't added it (I don't have a way to test it). I can add it if someone is willing to test it, or it can be added later.

fkokosinski commented:
> @fkokosinski Thanks, the issue is here: #61109

Thanks!

> I'm using RAIL without BT, so I haven't added it (I don't have a way to test it). I can add it if someone is willing to test it, or it can be added later.

What about `CONFIG_SOC_GECKO_USE_RAIL`? Do you not need to set it?

yonsch commented:
> What about CONFIG_SOC_GECKO_USE_RAIL? Do you not need to set it?

Yes, I do it in my `prj.conf`.
Also, two questions:
1. I see that the SDK already has a 4.3 version, is there any reason not to use it?
2. Currently the URLs in `module.yml` reference a branch and not a commit/tag. Wouldn't that be a problem if a newer version is pushed to that branch by Silabs?

fkokosinski commented:
> > What about CONFIG_SOC_GECKO_USE_RAIL? Do you not need to set it?
> 
> Yes, I do it in my `prj.conf`. Also, two questions:
> 
>     1. I see that the SDK already has a 4.3 version, is there any reason not to use it?

Not really. It just that newer blobs weren't tested with the BT driver.

>     2. Currently the URLs in `module.yml` reference a branch and not a commit/tag. Wouldn't that be a problem if a newer version is pushed to that branch by Silabs?

Good catch. We have SHAs of those blobs, so we can identify them, but indeed - it would be a if SiLabs were to push a newer version to that branch. Let's change URLs to point to tags instead.

yonsch committed:
soc: silabs_exx32: Select SOC_GECKO_SERIES1 for 1x devices

Select SOC_GECKO_SERIES1 for all Silabs devices of the series 1x.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>

yonsch committed:
pinctrl: silabs: Added default pinctrl for efr32xg13p

Added a default pinctrl for the efr32xg13p device.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>

yonsch commented:
Update: fixed FPU issue for series 1 devices and found out that pinctrl was not implemented for most of them. I've added an implementation for EFR32FG13P (BRD4255A which is what I have).

fkokosinski commented on boards/arm/efr32_radio/Kconfig.defconfig:
```diff
@@ -32,6 +32,7 @@ config LOG_BACKEND_SWO_FREQ_HZ
 if SOC_GECKO_USE_RAIL
 
 config FPU
+	default n if SOC_GECKO_SERIES1
```
Hmmm, we don't need to disable FPU if we're not using RAIL. So we should check if we're `SOC_GECKO_SERIES1` and if we're using RAIL here IMO.

yonsch replied:
This is already checked on line 32

fkokosinski replied:
Monday mornings ;) You're correct, thanks

yonsch committed:
pinctrl: silabs: Added default pinctrl for efr32xg1p

Added a default pinctrl for the efr32xg1p device.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>

yonsch committed:
pinctrl: silabs: Added default pinctrl for efr32xg12p

Added a default pinctrl for the efr32xg12p device.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>

fkokosinski commented:
Hi @yonsch! I've merged https://github.com/zephyrproject-rtos/hal_silabs/pull/35. Please update the HAL revision that this PR points to to the newest one. Thanks!

fabiobaltieri commented:
hey @yonsch looks like this broke some boards:

```
$ west build -p -b efm32wg_stk3800 samples/basic/blinky
...
CMake Error at /usr/local/google/home/fabiobaltieri/zephyrproject/modules/hal/silabs/gecko/CMakeLists.txt:54 (message):
  Unknown gecko series type.  Is CONFIG_SOC_GECKO_SERIES1/2 set?
```

could you take a look?
